pkg/reconciler/channel/channel.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package channel
+
+import (
+	"context"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	eventinginformers "github.com/knative/eventing/pkg/client/informers/externalversions/eventing/v1alpha1"
+	listers "github.com/knative/eventing/pkg/client/listers/eventing/v1alpha1"
+	"github.com/knative/eventing/pkg/logging"
+	"github.com/knative/eventing/pkg/reconciler"
+	"github.com/knative/pkg/controller"
+	"go.uber.org/zap"
+)
+
+const (
+	// ReconcilerName is the name of the reconciler
+	ReconcilerName = "Channels"
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName       = "channel-default-controller"
+	channelReconciled         = "ChannelReconciled"
+	channelUpdateStatusFailed = "ChannelUpdateStatusFailed"
+)
+
+type Reconciler struct {
+	*reconciler.Base
+
+	// listers index properties about resources
+	channelLister listers.ChannelLister
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+// NewController initializes the controller and is called by the generated code
+// Registers event handlers to enqueue events
+func NewController(
+	opt reconciler.Options,
+	channelInformer eventinginformers.ChannelInformer,
+) *controller.Impl {
+
+	r := &Reconciler{
+		Base:          reconciler.NewBase(opt, controllerAgentName),
+		channelLister: channelInformer.Lister(),
+	}
+	impl := controller.NewImpl(r, r.Logger, ReconcilerName, reconciler.MustNewStatsReporter(ReconcilerName, r.Logger))
+
+	r.Logger.Info("Setting up event handlers")
+	channelInformer.Informer().AddEventHandler(reconciler.Handler(impl.Enqueue))
+
+	return impl
+}
+
+// Reconcile will check if the channel is being watched by provisioner's channel controller
+// This will improve UX. See https://github.com/knative/eventing/issues/779
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
```

Comment:
Save the key into the logging context, then all the subsequent `logging.FromContext(ctx)` calls don't need to add the key.

Reply:
I understand but no need because there is only three places we log.